module-3/task2/slices/slice.go

```diff
@@ -1,21 +1,60 @@
 package slices
 
-func SortStringsInPlace(s []string) {
-}
+import "sort"
 
-func SortStringsPure(s []string) []string {
+func SortStringsInPlace(strings []string) {
+	sort.Strings(strings)
+}
 
+func SortStringsPure(strings []string) []string {
+	temp_strings := make([]string, len(strings))
+	copy(temp_strings, strings)
+	SortStringsInPlace(temp_strings)
+	return temp_strings
 }
 
 type User struct {
 	FirstName string
 	LastName  string
 }
 
-func SortUsersPure(s []User) []User {
-	return r
+// to implement sort.Interface
+type UserSlice []User
+
+func (users UserSlice) Len() int {
```

Comment:
I think this is not necessary to create own type that implements `sort.Interface` I'd prefer in this case use sort.Slice